docs: explain Kubernets resources used in Cyclone (#1337)
caicloud/cyclone: docs/crd.md

# Kubernetes Resources Used by Cyclone

As we known Cyclone is implemented with native Kubernetes resources with no extra dependencies. Following we will explain resources Kubernetes originally Defined and Cyclone Defined.

## Kubernetes Original Resources

* **Namespace**: Cyclone is a Multi-Tenant Isolation system. 

    In general, there are two namespaces for each tenant:

    * Metadata namespace: Store Metadata for one Tenant, and hold the tenant description information in annotations. *MUST* in the Control cluster(cluster cyclone running in), named in format of `cyclone-{tenant-name}`

>     * Workload namespace: Execute Workload. Maybe in the Control cluster OR other clusters. 

Reviewer: ```suggestion
    * Workload namespace: Where workloads execute. Can be in control cluster OR user clusters. 
```

>     However you can explicitly specify the two namespaces as one: `cyclone-{tenant-name}`, then Cyclone will store metadata and execute Workload in it(Not Recommend).

Reviewer: Explain the reason why not recommend to merge the two namespaces into one.

> * **Secret**: Cyclone supports [integrate external systems](./user_guide.md#integration-center).Cyclone uses `Secret` to protect the auth information of external systems as it is pretty sensitive.

Reviewer: ```suggestion
* **Secret**: Cyclone supports integrating [external systems](./user_guide.md#integration-center). Cyclone stores these external systems as `secrets` to protect their sensitive informations.
```
